### ISSUE TITLE
Stash 0.7.0 Cannot Have Unicode Prompt under Python 3

### ISSUE THREAD

**Issue by iamrecursion** [ywangd/stash#316]
I’ve always had a λ as my prompt (set in `.stashrc`). While this works fine under python 2, and continues to do so under the latest version, trying to launch stash under python 3 results in an error. It appears that the field is being decoded as ASCII. The traceback follows:

```python
Traceback (most recent call last):
  File "/private/var/mobile/Containers/Shared/AppGroup/AAD3B109-8A6C-429A-9498-C2D5902D5E24/Pythonista3/Documents/launch_stash.py", line 94, in <module>
    no_historyfile=ns.no_historyfile, command=ctp)
  File "/private/var/mobile/Containers/Shared/AppGroup/AAD3B109-8A6C-429A-9498-C2D5902D5E24/Pythonista3/Documents/site-packages/stash/core.py", line 145, in __init__
    self.runtime.load_rcfile(no_rcfile=no_rcfile)
  File "/private/var/mobile/Containers/Shared/AppGroup/AAD3B109-8A6C-429A-9498-C2D5902D5E24/Pythonista3/Documents/site-packages/stash/system/shruntime.py", line 122, in load_rcfile
    self.stash(ins.readlines(),
  File "/var/containers/Bundle/Application/98ACED98-EEA8-4091-AF58-F20A55A58DB4/Pythonista3.app/Frameworks/Py3Kit.framework/pylib/encodings/ascii.py", line 27, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xce in position 62: ordinal not in range(128)
```

**Comment by sn3ksoftware** [comment]
The problem is StaSh assumes that text should be decoded in ASCII, which is OK unless there are Unicode symbols. This is one thing that they have not fixed in the Python 3 compatibility (especially for scripts).
So, I added utf-8 encoding to the standard shruntime.py script.
This also allows you to run scripts with Unicode in them in StaSh’s Python 3 mode.
Copy and paste the code attached.
(The diff is below, if you want to see it.)
```
--- shruntime.py
+++ shruntime_n.py
@@ -118,10 +118,16 @@
 
         if not no_rcfile and os.path.exists(self.rcfile) and os.path.isfile(self.rcfile):
             try:
-                with open(self.rcfile) as ins:
-                    self.stash(ins.readlines(),
-                               persistent_level=1,
-                               add_to_history=False, add_new_inp_line=False)
+                if PY3:
+                    with open(self.rcfile, encoding="utf-8") as ins:
+                        self.stash(ins.readlines(),
+                                   persistent_level=1,
+                                   add_to_history=False, add_new_inp_line=False)
+                else:
+                    with open(self.rcfile) as ins:
+                        self.stash(ins.readlines(),
+                                   persistent_level=1,
+                                   add_to_history=False, add_new_inp_line=False)
             except IOError:
                 self.stash.write_message('%s: error reading rcfile\n' % self.rcfile)
 
@@ -539,12 +545,15 @@
         self.handle_PYTHONPATH()  # Make sure PYTHONPATH is honored
 
         try:
-            with open(file_path, "rU") as f:
-                content = f.read()
-                code = compile(
-                    content, file_path, "exec", dont_inherit=True
-                    )
-                exec(code, namespace, namespace)
+            if PY3:
+                with open(file_path, "rU", encoding="utf-8") as f:
+                    content = f.read()
+            else:
+                with open(file_path, "rU") as f:
+                    content = f.read()
+            code = compile(
+                content, file_path, "exec", dont_inherit=True)
+            exec(code, namespace, namespace)
             
             current_state.return_value = 0
```
[shruntime.txt.txt](https://github.com/ywangd/stash/files/3584176/shruntime.txt.txt)

**Comment by bennr01** [comment]
Fixed by @sn3ksoftware in PR #389.
